File: dayce/Logic/EventFilter.swift
import Foundation

enum EventFilter: String, CaseIterable, Identifiable {
    case all
    case countUp
    case countDown

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .all:
            return "All"
        case .countUp:
            return "Count Up"
        case .countDown:
            return "Countdown"
        }
    }

    func apply(to events: [DayEvent]) -> [DayEvent] {
        switch self {
        case .all:
            return events
        case .countUp:
            return events.filter { $0.kind == .countUp }
        case .countDown:
            return events.filter { $0.kind == .countDown }
        }
    }
}

File: android/app/src/main/java/com/motoish/dayce/ui/EventFilter.kt
package com.motoish.dayce.ui

import com.motoish.dayce.domain.DayEventKind

interface FilterableEvent {
    val id: Long
    val kind: DayEventKind
}

enum class EventFilter {
    All,
    CountUp,
    Countdown;

    fun <T : FilterableEvent> apply(events: List<T>): List<T> {
        return when (this) {
            All -> events
            CountUp -> events.filter { it.kind == DayEventKind.CountUp }
            Countdown -> events.filter { it.kind == DayEventKind.Countdown }
        }
    }
}
